File: Interfaces/IActivityService.cs
using depoWebAPI.Models;
using cms_webapi.DTOs;
using cms_webapi.Data;

namespace cms_webapi.Interfaces
{
    public interface IActivityService
    {
        Task<ApiResponse<List<ActivityDto>>> GetAllActivitiesAsync();
        Task<ApiResponse<ActivityDto>> GetActivityByIdAsync(long id);
        Task<ApiResponse<ActivityDto>> CreateActivityAsync(CreateActivityDto createActivityDto);
        Task<ApiResponse<ActivityDto>> UpdateActivityAsync(long id, UpdateActivityDto updateActivityDto);
        Task<ApiResponse<object>> DeleteActivityAsync(long id);
    }
}

File: WebApi/Interfaces/IActivityService.ts
import { ApiResponse } from '../Models/ApiResponse';
import { ActivityDto, CreateActivityDto, UpdateActivityDto } from '../Models/ActivityDto';

export interface IActivityService {
  getAllActivities(): Promise<ApiResponse<ActivityDto[]>>;
  getActivityById(id: number): Promise<ApiResponse<ActivityDto>>;
  createActivity(createActivityDto: CreateActivityDto): Promise<ApiResponse<ActivityDto>>;
  updateActivity(id: number, updateActivityDto: UpdateActivityDto): Promise<ApiResponse<ActivityDto>>;
  deleteActivity(id: number): Promise<ApiResponse<object>>;
}